src/index.scss

```diff
@@ -0,0 +1,305 @@
+$grid-gutter-width: 30px;
+$header-padding: 8px;
+$block-vertical-margin: 3px;
+$block-full-margin: $block-vertical-margin 0;
+$add-button-size: 34px;
+$add-button-font-size: 28px;
+$children-container-padding: $add-button-size / 2;
+$action-font-size: 15px;
+$add-transition-duration: 0.3s;
+$hover-transition-duration: 0.3s;
+$bounce-transition-timing: cubic-bezier(0.175, 0.885, 0.32, 1.275);
+$border-radius: 3px;
+$teal: #007d7e;
+$header-text-color-focus: #4d4d4d;
+$header-text-color: #888888;
+$block-border-color: #e6e6e6;
+$block-border-color-focus: #bbbbbb;
+$block-hover-background: #f5f5f5;
+$error-color: #cd3238;
+$error-border-color: #dbc7c8;
+$error-border-color-focus: #cdb2b3;
+$error-background-color: #fbefef;
+$screen-xs-max: 799px;
+
+.children-container {
+  position: relative;
+  padding: $children-container-padding 0
+           $children-container-padding $add-button-size;
+  transition: padding $hover-transition-duration ease-in-out;
+  @media (max-width: $screen-xs-max) {
+    padding-left: 0;
+    padding-top: $children-container-padding + $add-button-size;
+  }
+  &.is-dragging {
+    > button.add.visible {
+      opacity: 0;
+      pointer-events: none;
+    }
+  }
+  .block {
+    position: relative;
+    display: flex;
+    margin: $block-full-margin;
+    @media (max-width: $screen-xs-max) {
+      margin-bottom: $block-vertical-margin + $add-button-size;
+    }
+    &.has-error {
+      > .block-container {
+        border-color: $error-border-color;
+        > header {
+          background: $error-background-color;
+        }
+        > .content-container > .content {
+          border-color: $error-border-color;
+        }
+      }
+      &:hover, &:focus, &:focus-within {
```

Comment:
`:focus-within` still doesn't have that good browser support. Is this functionality important enough to justify adding a `--focus` class in JS, and add styles based on the class instead?

Reply:
That’s not worth doing, this is just a cosmetic functionality to make ancestors blocks with an light red background & border. And the errored blocks are all force opened, so there’s no way we can miss an error in a block, even without this focus-within.